tests/tools/invalidate/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+function(add_invalidation_test TEST_NAME)
+  set(options)
+  set(oneValueArgs STEP INPUT PIPELINE INVALIDATIONS)
+  set(multiValueArgs TARGETS)
+  cmake_parse_arguments(OPT "${options}" "${oneValueArgs}" "${multiValueArgs}"
+                        "${ARGN}")
+
+  add_test(
+    NAME "${TEST_NAME}"
+    COMMAND
+      bash -c "rm -rf ${CMAKE_BINARY_DIR}/${TEST_NAME} &&
+      mkdir ${CMAKE_BINARY_DIR}/${TEST_NAME} &&
+      ${CMAKE_BINARY_DIR}/libexec/revng/revng-pipeline \
+      -P ${OPT_PIPELINE} \
+      ${OPT_TARGETS} \
+      -i begin:Strings1:${OPT_INPUT} \
+      -l ${CMAKE_BINARY_DIR}/lib/librevngStringContainerLibrary.so \
+      -p ${CMAKE_BINARY_DIR}/${TEST_NAME} \
+      --step ${OPT_STEP} &&
+      ${CMAKE_BINARY_DIR}/libexec/revng/revng-invalidate \
+      -P ${OPT_PIPELINE} \
+      ${OPT_INVALIDATIONS} \
+      -l ${CMAKE_BINARY_DIR}/lib/librevngStringContainerLibrary.so \
+      -p ${CMAKE_BINARY_DIR}/${TEST_NAME} \
+      --step ${OPT_STEP}")
```

Comment:
I suspect `${CMAKE_BINARY_DIR}` is not needed.
I understand you want to be explicit, but I prefer to make command line looks approachable when debugging stuff.

Reply:
tried but it does not work, i think that the execution directory here is ${CMAKE_CURRENT_BINARY_DIR}